notebooks/3.jump-analysis/3.jump_analysis.py

```diff
@@ -0,0 +1,453 @@
+# Here, we used the International Chemical Identifier (InChI) to identify chemicals shared between the JUMP dataset and the Cell Injury dataset.
+
+# In[6]:
+
+
+cell_injury_InChI_keys = cell_injury_meta_df["Compound InChIKey"].unique().tolist()
+jump_InChI_keys = shared_jump_df["Metadata_InChIKey"].unique().tolist()
+
+# # identify common InChI Keys
+common_compounds_inchikey = list(
+    set(cell_injury_InChI_keys).intersection(jump_InChI_keys)
+)
+
+# # identify the compounds and display in cell injury data
+overlapping_compounds_df = cell_injury_meta_df.loc[
+    cell_injury_meta_df["Compound InChIKey"].isin(common_compounds_inchikey)
+]
+unique_compound_names = overlapping_compounds_df["Compound Name"].unique().tolist()
+print("Identified overlapping compounds:", ", ".join(unique_compound_names))
+
+# now create a dataframe where it contains the injury code, name and injury type
+overlapping_compounds_df = (
+    overlapping_compounds_df[
+        ["injury_code", "injury_type", "Compound Name", "Compound InChIKey"]
+    ]
+    .drop_duplicates()
+    .reset_index(drop=True)
+)
+
+# lower casing all the entries
+overlapping_compounds_df["injury_type"] = overlapping_compounds_df[
+    "injury_type"
+].str.lower()
+overlapping_compounds_df
+
+
+# Once the common compounds and their associated cell injury types are identified, the next step involves selecting it from the JUMP dataset to select only wells that possess the common InChI keys.
+
+# In[7]:
+
+
+# selecting rows that contains the overlapping compounds
+shared_treat_jump_df = shared_jump_df.loc[
+    shared_jump_df["Metadata_InChIKey"].isin(common_compounds_inchikey)
+]
+
+# augment filtered JUMP data with labels
+shared_treat_jump_df = pd.merge(
+    overlapping_compounds_df,
+    shared_treat_jump_df,
+    right_on="Metadata_InChIKey",
+    left_on="Compound InChIKey",
+)
+
+# shared treatment jump df
+print("shape: ", shared_treat_jump_df.shape)
+shared_treat_jump_df.head()
+
+
+# Now that we have identified the wells treated with overlapping treatments, we want to know the amount of wells that a specific treatment have.
+
+# In[8]:
+
+
+# count number of wells and agument with injury_code injury_yype and compound name
+well_counts_df = (
+    shared_treat_jump_df.groupby("Metadata_InChIKey")
+    # counting the numbver of wells
+    .size()
+    .to_frame()
+    .reset_index()
+    # merge based on InChIKey
+    .merge(
+        overlapping_compounds_df,
+        left_on="Metadata_InChIKey",
+        right_on="Compound InChIKey",
+    )
+    .drop(columns=["Compound InChIKey"])
+)
+
+# update columns
+well_counts_df.columns = [
+    "Metadata_InChIKey",
+    "n_wells",
+    "injury_code",
+    "injury_type",
+    "compund_name",
+]
+well_counts_df
+
+
+# In[9]:
+
+
+# this is our ground truth since this compound is also found in the cell-injury dataset
+jump_cyto_injury_df = shared_jump_df.loc[
+    shared_jump_df["Metadata_InChIKey"] == "IAKHMKGGTNLKSZ-INIZCTEOSA-N"
```

Comment:
what is it??

Reply:
Documentation updated! Here's the explanation:

The InChIKey "IAKHMKGGTNLKSZ-INIZCTEOSA-N" represents the compound found in both the cell-injury and JUMP-CP datasets.

Therefore the wells containing this compound (represented by the InChIKey) is then removed from the original JUMP-CP dataset and as a result, we used this wells as our ground truth.


Hopefully this made sense! Thanks!